src/pagination/components/Pagination.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+class Pagination extends React.Component{
+    constructor(props){
+        super(props);
+        this.state={
+            allDataRowsNumber: 0,
+            currentPage: 1,
+            dataPerPage: props.dataPerPage
+        }
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.setState({ allDataRowsNumber: nextProps.allDataRowsNumber });  
+    }
+
+    handleClick(event) {
+        event.preventDefault()
+        this.setState({
+          currentPage: Number(event.target.id)
```

Comment:
DOM details are coupled to our click handler in this case. Try to find a way to avoid that.